mlflow/recipes/steps/split.py

```diff
@@ -202,29 +240,46 @@ def _set_target_col_as_first(df, target_col):
                 "PROFILE", data_profile
             )
 
-            if self.positive_class:
-                positive_df, negative_df = (
-                    train_df[(mask := train_df[self.target_col] == self.positive_class)],
-                    train_df[~mask],
-                )
-
-                positive_negative_profile = get_pandas_data_profiles(
+            if self.task == "classification":
+                if self.positive_class is not None:
+                    mask = train_df[self.target_col] == self.positive_class
+                    dfs_for_profiles = [
+                        ("Positive", train_df[mask]),
+                        ("Negative", train_df[~mask]),
+                    ]
+                else:
+                    classes = np.unique(train_df[self.target_col])
+                    class_weights = compute_class_weight(
+                        class_weight="balanced",
+                        classes=classes,
+                        y=train_df[self.target_col],
+                    )
+                    class_weights = list(zip(classes, class_weights))
+                    class_weights = sorted(class_weights, key=lambda x: x[1], reverse=True)
+                    if len(class_weights) > 5:
```

Comment:
Can we add a parameter `_MAX_CLASSES_TO_PROFILE = 5` to the top of this file?

Reply:
Done.